### PR TITLE
[CPDEV-101741] - Support for new k8s versions

### PR DESCRIPTION
**Pull request by pranavcracker**
### Description
*  Adding support new k8s versions 1.27.13, 1.28.9, 1.29.4

### Test Cases
**TestCase 1**

Test Configuration:

- Hardware: Openstack
- OS: Ubuntu 22.04, CentOS 7
- Inventory: All-in-One, MiniHA

Steps:

1. **Kubemarine install** -- kubernetesVersion: _v1.28.9_
2. **Kubemarine upgrade** from _1.28.9_ to _1.29.4_
3. **kubemarine add/remove_node**

ER: Install, upgrade, add/remove node should succeed.

### Checklist
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] Integration CI passed
- [x] Unit tests. If Yes list of new/changed tests with brief description
- [x] There is no merge conflicts

### CODE REVIEW

File: kubemarine/resources/configurations/compatibility/kubernetes_versions.yaml [hunk 1]
```diff
@@ -137,12 +145,30 @@ compatibility_map:
...
     kubernetes-dashboard: v2.7.0
     local-path-provisioner: v0.0.26
     crictl: v1.29.0
+  v1.29.4:
+    calico: v3.28.0
+    nginx-ingress-controller: v1.10.0
+    kubernetes-dashboard: v2.7.0
+    local-path-provisioner: v0.0.26
+    crictl: v1.30.0
+  v1.30.1:
```
Comment (ilia1243) [inline kubemarine/resources/configurations/compatibility/kubernetes_versions.yaml:166]: To support v1.30.x, it is necessary to include code changes and respect comments from #647

Comment (pranavcracker) [reply kubemarine/resources/configurations/compatibility/kubernetes_versions.yaml]: removed v1.30.x from this PR

File: kubemarine/resources/configurations/compatibility/kubernetes_versions.yaml [hunk 2]
```diff
@@ -137,12 +143,24 @@ compatibility_map:
...
+    crictl: v1.30.0
   v1.29.1:
     calico: v3.27.0
     nginx-ingress-controller: v1.9.6
     kubernetes-dashboard: v2.7.0
     local-path-provisioner: v0.0.26
     crictl: v1.29.0
+  v1.29.4:
+    calico: v3.28.0
+    nginx-ingress-controller: v1.10.0
```
Comment (ilia1243) [inline kubemarine/resources/configurations/compatibility/kubernetes_versions.yaml:160]: v1.10.1 is available https://github.com/kubernetes/ingress-nginx/releases/tag/controller-v1.10.1

File: kubemarine/patches/software_upgrade.yaml [hunk 3]
```diff
@@ -8,8 +8,14 @@
...
-  calicoctl: []
-  crictl: []
+  calicoctl:
+  - v1.27.13
+  - v1.28.9
+  - v1.29.4
+  crictl:
+  - v1.27.13
+  - v1.28.9
+  - v1.29.4
```
Comment (ilia1243) [inline kubemarine/patches/software_upgrade.yaml:18]: These are new versions. There is nothing to upgrade for them

Comment (pranavcracker) [reply kubemarine/patches/software_upgrade.yaml]: updated

File: kubemarine/plugins/yaml/nginx-ingress-controller-v1.10.1-original.yaml [hunk 4]
```diff
@@ -0,0 +1,662 @@
...
+        - /nginx-ingress-controller
+        - --publish-service=$(POD_NAMESPACE)/ingress-nginx-controller
+        - --election-id=ingress-nginx-leader
+        - --controller-class=k8s.io/ingress-nginx
+        - --ingress-class=nginx
+        - --configmap=$(POD_NAMESPACE)/ingress-nginx-controller
+        - --validating-webhook=:8443
+        - --validating-webhook-certificate=/usr/local/certificates/cert
+        - --validating-webhook-key=/usr/local/certificates/key
+        - --enable-metrics=false
```
Comment (ilia1243) [inline kubemarine/plugins/yaml/nginx-ingress-controller-v1.10.1-original.yaml:434]: They disabled metrics by default. Please investigate how this affect exposition of metrics #51

**Review by alexarefev — APPROVED** [review]

File: kubemarine/plugins/nginx_ingress.py [hunk 5]
```diff
@@ -247,7 +247,7 @@ def enrich_deamonset_ingress_nginx_controller_container(self, manifest: Manifest
             extra_args.append('--default-ssl-certificate' + '=' + 'kube-system/default-ingress-cert')
 
         self.enrich_deamonset_ingress_nginx_controller_container_args(
-            manifest, remove_args=['--publish-service'], extra_args=extra_args)
+            manifest, remove_args=['--publish-service','--enable-metrics'], extra_args=extra_args)
```
Comment (ilia1243) [inline kubemarine/plugins/nginx_ingress.py:250]: This may be covered by unit tests.

**Review by ilia1243 — APPROVED** [review]

File: kubemarine/plugins/yaml/nginx-ingress-controller-v1.10.1-original.yaml (hunk 4 above)
Comment (ilia1243) [reply kubemarine/plugins/yaml/nginx-ingress-controller-v1.10.1-original.yaml:434]: Fixed by https://github.com/Netcracker/KubeMarine/pull/662/commits/f696fa38ea386727fdc760ff99e3b1428c109494

Comment (pranavcracker) [reply kubemarine/plugins/yaml/nginx-ingress-controller-v1.10.1-original.yaml:434]: addressed through enrichment.

File: kubemarine/plugins/nginx_ingress.py (hunk 5 above)
Comment (pranavcracker) [reply kubemarine/plugins/nginx_ingress.py:250]: unit test updated.

**Review by koryaga — APPROVED** [review]

**Pull request merged**